morango/syncsession.py

```diff
@@ -0,0 +1,407 @@
+            merge_conflict_store = """REPLACE INTO {store} (id, serialized, deleted, last_saved_instance, last_saved_counter, model_name,
+                                                            profile, partition, conflicting_serialized_data, dirty_bit)
+                                                SELECT store.id, store.serialized, store.deleted, '{current_instance_id}', {current_instance_counter},
+                                                       store.model_name, store.profile, store.partition,
+                                                       buffer.serialized || '\n' || store.conflicting_serialized_data, 1
+                                                FROM {buffer} AS buffer, {rmcb} AS rmcb, {store} AS store, {rmc} as rmc
+                                    /*Scope to a single record.*/
+                                    WHERE store.id = rmcb.model_uuid
+                                    AND store.id = rmc.store_model_id
+                                    AND store.id = buffer.model_uuid
+                                    /*Where buffer rmc is greater than store rmc*/
+                                    AND rmcb.instance_id = rmc.instance_id
+                                    AND rmcb.counter > rmc.counter
+                                    AND rmcb.transfer_session_id = '{transfer_session_id}'
+                                    AND buffer.transfer_session_id = '{transfer_session_id}'
+                                    /*Exclude fast-forwards*/
+                                    AND NOT EXISTS (SELECT 1 FROM {rmcb} AS rmcb2 WHERE store.id = rmcb2.model_uuid
+                                                                                  AND store.last_saved_instance = rmcb2.instance_id
+                                                                                  AND store.last_saved_counter <= rmcb2.counter
+                                                                                  AND rmcb2.transfer_session_id = '{transfer_session_id}')
+                                          """.format(buffer=Buffer._meta.db_table,
+                                                     rmcb=RecordMaxCounterBuffer._meta.db_table,
+                                                     store=Store._meta.db_table,
+                                                     rmc=RecordMaxCounter._meta.db_table,
+                                                     transfer_session_id=self.transfer_session_id,
+                                                     current_instance_id=current_id.id,
+                                                     current_instance_counter=current_id.counter)
+            cursor.execute(merge_conflict_store)
+
+    def _dequeuing_update_rmcs_last_saved_by(self, cursor, current_id):
+
+            # updated or create rmc for merge conflicts with id of current system
+            merge_conflict_store = """REPLACE INTO {rmc} (instance_id, counter, store_model_id)
```

Comment:
Yes, if it violates the `unique_together` constraint it will replace that row with the new values. In this case, the new counter.